sdk-tools/src/sign.rs

```diff
@@ -0,0 +1,272 @@
+        &mut self,
+        unsigned_enclave: &Path,
+        config_path: &Path,
+        output_datfile: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("gendata")
+            .arg("-enclave")
+            .arg(unsigned_enclave)
+            .arg("-config")
+            .arg(config_path)
+            .arg("-out")
+            .arg(output_datfile);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
+
+    /// Combine an unsigned enclave and signature into the output enclave, after
+    /// checking the signature.
+    pub fn catsig(
+        &mut self,
+        unsigned_enclave: &Path,
+        config_path: &Path,
+        public_key_pem: &Path,
+        gendata_output: &Path,
+        signature: &Path,
+        output_enclave: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("catsig")
+            .arg("-enclave")
+            .arg(unsigned_enclave)
+            .arg("-config")
+            .arg(config_path)
+            .arg("-key")
+            .arg(public_key_pem)
+            .arg("-unsigned")
+            .arg(gendata_output)
+            .arg("-sig")
+            .arg(signature)
+            .arg("-out")
+            .arg(output_enclave);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        cmd
+    }
+
+    /// Examine a signed enclave file and dump the data
+    pub fn dump(
+        &mut self,
+        signed_enclave: &Path,
+        css_file_path: &Path,
+        dump_file_path: &Path,
+    ) -> Command {
+        let mut cmd = Command::new(self.sgx_sign_path.clone());
+        cmd.arg("dump")
+            .arg("-enclave")
+            .arg(signed_enclave)
+            .arg("-dumpfile")
+            .arg(dump_file_path)
+            .arg("-cssfile")
+            .arg(css_file_path);
+
+        if self.ignore_relocation_error {
+            cmd.arg("-ignore-rel-error");
+        }
+
+        if self.ignore_init_section_error {
+            cmd.arg("-ignore-init-sec-error");
+        }
+
+        if self.resign {
+            cmd.arg("-resign");
+        }
+
+        cmd
+    }
+}
+
+/// Construct a new SgxSign utility around the given executable path
+impl From<PathBuf> for SgxSign {
+    fn from(sgx_sign_path: PathBuf) -> Self {
+        Self {
+            sgx_sign_path,
+            ignore_relocation_error: false,
+            ignore_init_section_error: false,
+            resign: false,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_generate_sign_command() {
```

Comment:
:upside_down_face: nit suggest omitting "test", other functions too.
```suggestion
    fn generate_sign_command() {
```